### DISCUSSION TITLE
Brainstorming: using Trice and Serial Studio together

### DISCUSSION THREAD

**Discussion by escherstair** [rokath/trice#288] [Ideas]
I don't know if someone knows and/or uses [Serial Studio](https://github.com/Serial-Studio/Serial-Studio).
[This blog post](https://www.alex-spataru.com/blog/introducing-serial-studio) explains the reasons for the project.

Basically it uses a JSON file on the PC (with all the immutable data, and some placeholders `%1`, `%2`, `%3`, ...), and the microcontroller sends over UART only the measurements values.
Serial Studio requires that the microcontroller senda a CSV string with the values.

When I read this, I started thinking....
It's not full clear to me, at this moment, but is there space for using Trice and Serial Studio together?
Add to Serial Studio support for (a subset of) Trice frames?
Add to trice tool an option to send data in Serial Studio format?

It's only brainstorming, at the moment

**Comment by rokath** [comment]
You wrote: "Serial Studio requires that the microcontroller senda a CSV string with the values.". This seems to make things easy: `TRICE( "csv:%d," v0 );` or `TRICE( "csv:%f,%f,%f,%f\n", aFloat(v0), aFloat(v1), aFloat(v2), aFloat(v3) );`.  You could start with simply `TRICE( "%d," v0 );` messages and give them to the Serila Studio using the  `trice -logfile` option. 

Maybe this works too: `trice l -p COMx -logfile trice.txt` and then `cat trice.txt | grep "csv:" | cut -b 55- > ss.txt` and then give ss.txt to serial studio. You may need to remove the newlines as well.
